app/models/work_url.rb

```diff
@@ -0,0 +1,74 @@
+class WorkUrl < ApplicationRecord
+  belongs_to :work
+
+  VARIANTS = %w[original minimal no_www with_www encoded decoded minimal_no_protocol_no_www].freeze
+
+  def self.build_url_attributes(url)
+    url = UrlFormatter.new(url)
+
+    VARIANTS.to_h { |s| [s.to_sym, url.send(s)] }
+  end
+
+  def self.prepare(work, url)
+    work_url = WorkUrl.find_or_initialize_by(work_id: work.id)
+
+    work_url.assign_attributes(self.build_url_attributes(url))
+
+    work.work_url = work_url
+
+    work_url
+  end
+
+  def self.find_by_url_generation_key
```

Comment:
Can we make most of these `private` to avoid other classes calling them that don't need to? (Similar question from for the method `build_url_attributes` above.)

Reply:
Actually, I am wondering if we will need them long term (e.g. can the caching be removed once we get everything converted to the new table?)